### PR TITLE
Preserving language and indentation of code block when serialized/deserialized

### PR DESCRIPTION
**Pull request by ebads67**
- The language and indentation of code block is preserved when serialized/deserialized

### CODE REVIEW

**Comment by ivailop7** [comment]
@acywatson can we get your thoughts here?

File: packages/lexical/src/nodes/LexicalTextNode.ts [hunk 1]
```diff
@@ -1128,6 +1128,7 @@ const preParentCache = new WeakMap<Node, null | Node>();
 function isNodePre(node: Node): boolean {
   return (
     node.nodeName === 'PRE' ||
+    node.nodeName === 'CODE' ||
```
Comment (zurfyx) [inline packages/lexical/src/nodes/LexicalTextNode.ts:1131]: This is technically incorrect, doesn't follow the W3C spec but given that this is Lexical's fault for not wrapping code in `pre` I believe it's OK to have..

**Review by zurfyx — COMMENTED** [review]
LGTM! I can see it's still a draft, is it ready to stamp?

**Comment by ivailop7** [comment]
> LGTM! I can see it's still a draft, is it ready to stamp?

Yes, Ebad wanted you guys to have a look before converting for Ready for Review. Have switched it from Draft to Ready now. Thanks Gerard!

**Review by zurfyx — APPROVED** [review]

**Pull request merged**